api/v1/aerospikecluster_validating_webhook.go

```diff
@@ -2301,3 +2306,61 @@ func (c *AerospikeCluster) validateMaxUnavailable() error {
 
 	return nil
 }
+
+func (c *AerospikeCluster) validateEnableDynamicConfigUpdate() error {
+	if !GetBool(c.Spec.EnableDynamicConfigUpdate) {
+		return nil
+	}
+
+	if len(c.Status.Pods) == 0 {
+		return nil
+	}
+
+	minInitVersion, err := getMinRunningInitVersion(c.Status.Pods)
+	if err != nil {
+		return err
+	}
+
+	val, err := lib.CompareVersions(minInitVersion, minInitVersion4DynamicConf)
+	if err != nil {
+		return fmt.Errorf("failed to check image version: %v", err)
+	}
+
+	if val < 0 {
+		return fmt.Errorf("cannot enable enableDynamicConfigUpdate flag, init container are running version less than %s",
+			minInitVersion4DynamicConf)
+	}
+
+	return nil
+}
+
+func getMinRunningInitVersion(pods map[string]AerospikePodStatus) (string, error) {
+	minVersion := ""
+
+	for idx := range pods {
+		if pods[idx].InitImage != "" {
+			version, err := GetImageVersion(pods[idx].InitImage)
+			if err != nil {
+				return "", err
+			}
+
+			if minVersion == "" {
+				minVersion = version
+				continue
+			}
+
+			val, err := lib.CompareVersions(version, minVersion)
+			if err != nil {
+				return "", fmt.Errorf("failed to check image version: %v", err)
+			}
+
+			if val < 0 {
+				minVersion = version
+			}
+		} else {
+			return baseInitVersion, nil
+		}
+	}
+
+	return minVersion, nil
+}
```

Comment:
Instead of adding new var like minVersion and baseInitVersion, how about passing the version to be compared as a param and return true or false based on init tag comparison. Maybe move this code in `validateEnableDynamicConfigUpdate` func itself

Reply:
Both implementations are fine, just wanted to give a generic function to return the minimum init version running in the cluster, which may be helpful in the future.